frontend/src/components/MinioArtifactPreview.test.tsx

```diff
@@ -0,0 +1,224 @@
+          <a
+            class="link"
+            rel="noopener"
+            target="_blank"
+          />
+          <div
+            class="preview"
+          >
+            <small>
+              <pre>
+                preview ...
+              </pre>
+            </small>
+          </div>
+        </div>
+      </div>
+    `);
+  });
+
+  it('handles minio artifact', async () => {
+    const minioArtifact = {
+      accessKeySecret: { key: 'accesskey', optional: false, name: 'minio' },
+      bucket: 'foo',
+      endpoint: 'minio.kubeflow',
+      key: 'bar',
+      secretKeySecret: { key: 'secretkey', optional: false, name: 'minio' },
+    };
+    const container = document.body.appendChild(document.createElement('div'));
+    await act(async () => {
+      render(<MinioArtifactPreview value={minioArtifact} />, { container });
+    });
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <div
+          class="root"
+        >
+          <a
+            class="link"
+            rel="noopener"
+            target="_blank"
+          />
+          <div
+            class="preview"
+          >
+            <small>
+              <pre>
+                preview ...
+              </pre>
+            </small>
+          </div>
+        </div>
+      </div>
+    `);
+  });
+
+  it('handles artifact with namespace', async () => {
+    const minioArtifact = {
+      accessKeySecret: { key: 'accesskey', optional: false, name: 'minio' },
+      bucket: 'foo',
+      endpoint: 'minio.kubeflow',
+      key: 'bar',
+      secretKeySecret: { key: 'secretkey', optional: false, name: 'minio' },
+    };
+    const { container } = render(
+      <MinioArtifactPreview value={minioArtifact} namespace='namespace' />,
+    );
+    await act(TestUtils.flushPromises);
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <div
+          class="root"
+        >
+          <a
+            class="link"
+            rel="noopener"
+            target="_blank"
+          />
+          <div
+            class="preview"
+          >
+            <small>
+              <pre>
+                preview ...
+              </pre>
+            </small>
+          </div>
+        </div>
+      </div>
+    `);
+  });
+
+  it('handles artifact cleanly even when fetch fails', async () => {
+    const minioArtifact = {
+      accessKeySecret: { key: 'accesskey', optional: false, name: 'minio' },
+      bucket: 'foo',
+      endpoint: 'minio.kubeflow',
+      key: 'bar',
+      secretKeySecret: { key: 'secretkey', optional: false, name: 'minio' },
+    };
+    readFile.mockRejectedValue('unknown error');
+    const { container } = render(<MinioArtifactPreview value={minioArtifact} />);
+    await act(TestUtils.flushPromises);
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <div
+          class="root"
+        >
+          <a
+            class="link"
+            rel="noopener"
+            target="_blank"
+          />
```

Comment:
All the <a/>s should have href and content, is this a bug?

Reply:
yes. I missed this too. Let me check

Reply:
found the bug. cuz I mocked the entire `APIs` module so the functions did not return anything. Removed the unneccessary mocks. Works now.